### DISCUSSION TITLE
Recommended Software Versions for AirSim Python Environment

### DISCUSSION THREAD

**Discussion by lyvelyte** [microsoft/AirSim#3970] [Support Q&A]
Little bit of context: We've been developing with AirSim v1.4.0 for 6 months or so. Wonderful tool. 

Unfortunately we have been running into some issues with the segmented image not matching the provided color pallet. The infrared image also doesn't seem to update if object IDs are set manually. Both of these combined result in no way to get segmentation object IDs for an Image that match what is set. 

I believe some  of these bugs have already been fixed so we went to upgrade AirSim to 1.5.0. Unfortunately we run into rcp_errors. 

The documentation for python setup is a bit vague and most certainly not up to date in all places. Could someone please provide a list of version numbers, etc. that they are using with the latest source code on windows? Specifically, it would be wonderful to know the recommended versions for the following to help get everything setup and running smoothly for myself and others:

**Requested Software Version List:**
_General_
- Unreal Engine Version 
- OS Version (e.g. Window 10 x64)
- Anaconda 
- CUDA Version

_Conda Environment:_
- Python
- pip
- msgpack-rpc-python
- airsim (the python library)

_Visual Studio 2019:_
- Cmake
- Windows 10 SDK (Do we only need 1 installed? Can we use the latest?)
- .net SDK (Pretty sure this is required to build and run Blocks but did not see mentioned in documentation)
- Any other components that are necessary for building and running latest Blocks with AirSim. 

Thank you to any and all who may be able to offer up some or all of this information! Perhaps we could work to maintain some sort of compatibility list in the future.

**Comment by jonyMarino** [comment]
Hi @Iyvelyte! please run `pip install airsim --upgrade` to use the python client with the current master.

**Comment by lyvelyte** [comment]
Hi @jonyMarino! Thanks! I wasn't sure if pip was fetching the latest version so I built and copied the airsim python client + setup_path.py + added the import to my working directory and copied the plugin to my unreal project. I was still having some issues and figured out that one of my issues was actually just a bad settings.json. I was defining some invalid cameras in computer vision mode. I can now run the latest master, thanks again!

However, now I've got a new issue! Maybe I should make a new post... Now I am trying to retexture objects using simSwapTextures. I've been able to change the parent class of my object to TextureShuffleActor and it has gained the member DynamicMaterial however the TextureSwappableMaterial is missing! Thanks for any help with this issue!

**Reply by jonyMarino** [reply]
It is a pleasure to help. Better to open a new discussion. It is not my area of expertise.
